helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -432,6 +479,17 @@ protected Map<String, ResourceAssignment> emergencyRebalance(
 
     // Step 3: persist result to metadata store
     persistBestPossibleAssignment(newAssignment);
+
+    // Step 4: handle delayed rebalance minActiveReplica
+    // Note this result is one step branching from the main calculation and SHOULD NOT be persisted -- it is temporary,
+    // and only apply during the delayed window of those offline yet active nodes, a definitive resolution will happen
+    // once the node comes back of remain offline after the delayed window.
+    Map<String, ResourceAssignment> assignmentWithDelayedRebalanceAdjust = newAssignment;
+    if (_partialRebalanceRunner.isAsyncPartialRebalanceEnabled()) {
+      assignmentWithDelayedRebalanceAdjust =
+          handleDelayedRebalanceMinActiveReplica(clusterData, resourceMap, activeNodes, newAssignment, algorithm);
+    }
```

Comment:
We changed the check condition. Previous one means if we have delayed nodes, then we perform the recompute, now it is if delay is enabled then we just compute it. It would be adding more computing time.

Reply:
No, the check is still there, inside `handleDelayedRebalanceMinActiveReplica`, if it's not needed, the current assignment is returned.